injector/src/main/java/edu/ucr/cs/riple/injector/location/OnLocalVariable.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2023 Nima Karimipour
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package edu.ucr.cs.riple.injector.location;
+
+import edu.ucr.cs.riple.injector.Helper;
+import java.nio.file.Path;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+/**
+ * Represents a location for local variable element. This location is used to apply changes to a
+ * local variable.
+ *
+ * <p>Annotations on local variables will be applied on the declaration and all its type arguments
+ * if exists. For instance: {@code Foo<Bar, Map<Bar, Bar>> baz}
+ *
+ * <p>Will be annotated as: {@code @Annot Foo<@Annot Bar, @Annot Map<@Annot Bar, @Annot Bar>> baz}
```

Comment:
Is the logic for injecting these annotations in this class, or somewhere else?  I don't see it in this class.

Reply:
Sorry for that. This is left over before splitting this PR into two separate PRs. Annotation injections on type arguments has been implemented in #187. It also not limited to just local variables. Fixed the javadoc f1ffe0808aca8e184ebb233863ecf67fd074881e